sdks/v4-sdk/src/entities/trade.ts

```diff
@@ -283,56 +276,19 @@ export class Trade<TInput extends Currency, TOutput extends Currency, TTradeType
     }[],
     tradeType: TTradeType
   ): Promise<Trade<TInput, TOutput, TTradeType>> {
-    const populatedRoutes: {
+    const swaps: {
       route: Route<TInput, TOutput>
       inputAmount: CurrencyAmount<TInput>
       outputAmount: CurrencyAmount<TOutput>
-    }[] = []
-
-    for (const { route, amount } of routes) {
-      const amounts: CurrencyAmount<Token | Currency>[] = new Array(route.currencyPath.length)
-      let inputAmount: CurrencyAmount<TInput>
-      let outputAmount: CurrencyAmount<TOutput>
-
-      if (tradeType === TradeType.EXACT_INPUT) {
-        invariant(amount.currency.equals(route.input), 'INPUT')
```

Comment:
we probably want to keep this check and the one on line 314

Reply:
The checks are preserved in `fromRoute`.

Reply:
L😭